docs/src/ffi-error-handling.md

```diff
@@ -0,0 +1,249 @@
+Postgres is written in C.  pgrx is written in Rust.  Between them is a boundary where each blindly believes the other
+behaves in an expected way.  Our primary concern with this boundary is error handling.
+
+There are many other concerns across this boundary such as function call ABIs and pointer ownership, but these are generally
+"obvious" concerns to anyone that's done any FFI development and won't be discussed here in detail.  
+
+
+# High-level Postgres Error Handling Overview
```

Comment:
We should probably mention that this is describing the workings behind `ereport`, as C programmers are more likely to recognize that name.